### PR TITLE
Call an init function from config modules.

### PR DESCRIPTION
**Pull request by oliverchang**
Modules can define a "module_init" module containing a "appengine"
function which is called on App Engine startup.

Also expose a way to register issue tracker implementations.

### CODE REVIEW

File: src/appengine/libs/issue_management/issue_tracker_utils.py [hunk 1]
```diff
@@ -22,6 +22,14 @@
...
 
 _ISSUE_TRACKER_CACHE_CAPACITY = 8
+_ISSUE_TRACKER_CONSTRUCTORS = {
+    'monorail': monorail.get_issue_tracker,
+}
+
+
+def register_issue_tracker(tracker_type, constructor):
+  """Register an issue tracker implementation."""
+  _ISSUE_TRACKER_CONSTRUCTORS[tracker_type] = constructor
```
Comment (inferno-chromium) [inline src/appengine/libs/issue_management/issue_tracker_utils.py]: nit: Should we add something to skip registration if exists already ?

File: src/appengine/libs/issue_management/issue_tracker_utils.py [hunk 2]
```diff
@@ -43,10 +51,12 @@ def get_issue_tracker(project_name=None):
   if not issue_project_config:
     raise ValueError('Issue tracker for {} does not exist'.format(project_name))
 
-  if issue_project_config['type'] == 'monorail':
-    return monorail.get_issue_tracker(project_name)
+  constructor = _ISSUE_TRACKER_CONSTRUCTORS.get(issue_project_config['type'])
+  if not constructor:
+    raise ValueError('Invalid issue tracker type ' +
```
Comment (inferno-chromium) [inline src/appengine/libs/issue_management/issue_tracker_utils.py]: nit: colon after type, type:

Comment (oliverchang) [reply src/appengine/libs/issue_management/issue_tracker_utils.py]: done.

File: src/appengine/libs/issue_management/issue_tracker_utils.py (hunk 1 above)
Comment (oliverchang) [reply src/appengine/libs/issue_management/issue_tracker_utils.py]: made this raise an exception. should only be registered once.

**Pull request merged**